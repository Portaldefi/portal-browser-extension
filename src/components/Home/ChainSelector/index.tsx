import React, { useMemo, useCallback } from 'react';
import { Dropdown, DropdownProps } from 'semantic-ui-react';

import { useAppDispatch, useAppSelector } from '../../../hooks';
import { selectChain } from '../../../slices/keySlice';
import { cutter } from '@utils/text';
import chains from '@/config/chains';

export default () => {
  const dispatch = useAppDispatch();
  const selectedChain = useAppSelector(state => state.key.selectedChain);

  const chainOptions = useMemo(() => chains.map((chain, idx) => ({
    key: chain.name,
    value: chain.name,
    text: chain.name.toUpperCase()
  })), []);

  const onChainSelect = useCallback((event: React.SyntheticEvent<HTMLElement>, data: DropdownProps) => {
    dispatch(selectChain(data.value as string));
  }, []);


  return (
    <Dropdown
      className='chain-selector'
      selection
      options={chainOptions}
      value={selectedChain || chainOptions[0]?.value}
      onChange={onChainSelect}
    />
  );
}
