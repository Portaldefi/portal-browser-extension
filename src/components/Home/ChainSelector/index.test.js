import ChainSelector from './index';
import { create, act } from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux';

describe('ChainSelector', () => {
  const initialState = {
    menu: {
      items: []
    },
    key: {
      privateKey: '',
      privateExtendedKey: '',
      identity: [
        [{
          address: 'test address'
        }]
      ],
      selectedIdentityId: 0,
      selectedChainId: 0
    }
  };
  const mockStore = configureStore();
  let store;
  it('should work', () => {
    let tree;
    act(() => {
      store = mockStore(initialState);
      tree = create(
        <Provider store={store}>
          <ChainSelector />
        </Provider>
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});