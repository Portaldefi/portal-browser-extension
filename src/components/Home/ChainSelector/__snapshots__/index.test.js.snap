// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChainSelector should work 1`] = `
<div
  aria-expanded={false}
  className="ui selection dropdown chain-selector"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="listbox"
  tabIndex={0}
>
  <div
    aria-atomic={true}
    aria-live="polite"
    className="divider text"
    role="alert"
  >
    ETHEREUM
  </div>
  <i
    aria-hidden="true"
    className="dropdown icon"
    onClick={[Function]}
  />
  <div
    className="menu transition"
  >
    <div
      aria-checked={true}
      aria-selected={true}
      className="active selected item"
      onClick={[Function]}
      role="option"
      style={
        {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        ETHEREUM
      </span>
    </div>
    <div
      aria-checked={false}
      aria-selected={false}
      className="item"
      onClick={[Function]}
      role="option"
      style={
        {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        BITCOIN
      </span>
    </div>
    <div
      aria-checked={false}
      aria-selected={false}
      className="item"
      onClick={[Function]}
      role="option"
      style={
        {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        COSMOS
      </span>
    </div>
    <div
      aria-checked={false}
      aria-selected={false}
      className="item"
      onClick={[Function]}
      role="option"
      style={
        {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        BINANCE
      </span>
    </div>
    <div
      aria-checked={false}
      aria-selected={false}
      className="item"
      onClick={[Function]}
      role="option"
      style={
        {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        POLYGON
      </span>
    </div>
  </div>
</div>
`;
